[2024-11-05 18:01:11,583][root][INFO] - ---------------------------------------------------------------
[2024-11-05 18:01:11,584][root][INFO] - Env:
CUDA_VISIBLE_DEVICES: null
PYTHONOPTIMIZE: null

[2024-11-05 18:01:11,585][root][INFO] - Command line args:

[2024-11-05 18:01:11,585][root][INFO] - ---------------------------------------------------------------
[2024-11-05 18:01:11,587][root][INFO] - Run config:
device: cuda:4
solver_fn: nk.optimizer.solver.cholesky
lr: 0.0001
diag_shift: 1.0e-10
n_iter: 2000
chunk_size_jac: 2
chunk_size_vmap: 1
save_every: 3
base_dir: /scratch/.amisery/grad_sample/
model:
  _target_: grad_sample.models.ising.TFI
  L: 3
  h: 1.0
ansatz:
  _target_: nk.models.RBM
  alpha: 1
task:
  _target_: grad_sample.tasks.fullsum_train.Trainer

[2024-11-05 18:01:11,588][root][INFO] - ---------------------------------------------------------------
[2024-11-05 18:01:14,109][root][CRITICAL] - Error in call to target 'grad_sample.tasks.fullsum_train.Trainer':
InstantiationException('Error in call to target \'grad_sample.models.ising.TFI\':\nRuntimeError("Unable to initialize backend \'cuda\': FAILED_PRECONDITION: No visible GPU devices. (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)")\nfull_key: model')
full_key: task
Traceback (most recent call last):
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/xla_bridge.py", line 885, in backends
    backend = _init_backend(platform)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/xla_bridge.py", line 971, in _init_backend
    backend = registration.factory()
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/xla_bridge.py", line 671, in factory
    return xla_client.make_c_api_client(plugin_name, updated_options, None)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jaxlib/xla_client.py", line 200, in make_c_api_client
    return _xla.get_c_api_client(plugin_name, options, distributed_client)
jaxlib.xla_extension.XlaRuntimeError: FAILED_PRECONDITION: No visible GPU devices.

During handling of the above exception, another exception occurred:

jax.errors.SimplifiedTraceback: For simplicity, JAX has removed its internal frames from the traceback of the following exception. Set JAX_TRACEBACK_FILTERING=off to include these.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 92, in _call_target
    return _target_(*args, **kwargs)
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/models/ising.py", line 16, in __init__
    self.H_jax = self.H.to_jax_operator()
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/netket/operator/_ising/numba.py", line 99, in to_jax_operator
    return IsingJax(
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/netket/operator/_ising/jax.py", line 58, in __init__
    J = jnp.array(J, dtype=dtype)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/numpy/lax_numpy.py", line 4086, in array
    out_array: Array = lax_internal._convert_element_type(
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/lax/lax.py", line 571, in _convert_element_type
    return convert_element_type_p.bind(
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/lax/lax.py", line 2609, in _convert_element_type_bind
    operand = core.Primitive.bind(convert_element_type_p, operand,
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/core.py", line 439, in bind
    return self.bind_with_trace(find_top_trace(args), args, params)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/core.py", line 443, in bind_with_trace
    out = trace.process_primitive(self, map(trace.full_raise, args), params)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/core.py", line 949, in process_primitive
    return primitive.impl(*tracers, **params)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/jax/_src/dispatch.py", line 88, in apply_primitive
    outs = fun(*args)
RuntimeError: Unable to initialize backend 'cuda': FAILED_PRECONDITION: No visible GPU devices. (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 92, in _call_target
    return _target_(*args, **kwargs)
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/fullsum_train.py", line 16, in __init__
    super().__init__(cfg)
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/tasks/base.py", line 22, in __init__
    self.model = instantiate(self.cfg.model)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 226, in instantiate
    return instantiate_node(
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 347, in instantiate_node
    return _call_target(_target_, partial, args, kwargs, full_key)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 97, in _call_target
    raise InstantiationException(msg) from e
hydra.errors.InstantiationException: Error in call to target 'grad_sample.models.ising.TFI':
RuntimeError("Unable to initialize backend 'cuda': FAILED_PRECONDITION: No visible GPU devices. (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)")
full_key: model

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/amisery/phd/tfi_gs_samples/grad_sample/src/grad_sample/main.py", line 49, in main
    task = instantiate(cfg.task, cfg)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 226, in instantiate
    return instantiate_node(
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 347, in instantiate_node
    return _call_target(_target_, partial, args, kwargs, full_key)
  File "/scratch/.amisery/netket2/lib/python3.10/site-packages/hydra/_internal/instantiate/_instantiate2.py", line 97, in _call_target
    raise InstantiationException(msg) from e
hydra.errors.InstantiationException: Error in call to target 'grad_sample.tasks.fullsum_train.Trainer':
InstantiationException('Error in call to target \'grad_sample.models.ising.TFI\':\nRuntimeError("Unable to initialize backend \'cuda\': FAILED_PRECONDITION: No visible GPU devices. (you may need to uninstall the failing plugin package, or set JAX_PLATFORMS=cpu to skip this backend.)")\nfull_key: model')
full_key: task
